tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml

```diff
@@ -0,0 +1,131 @@
+summary: Test that snapd with testkeys on UC20 can boot a model with grade signed.
+
+systems: [ubuntu-20.04-64]
+
+environment:
+  # use tpm + secure boot to get full disk encryption, this is explicitly needed
+  # for grade: secured
+  NESTED_ENABLE_TPM: true
+  NESTED_ENABLE_SECURE_BOOT: true
+
+  # kvm doesn't work with tpm + secure boot on gce unfortunately
+  NESTED_ENABLE_KVM: false
+
+  # use snapd from the spread run so that we have testkeys trusted in the snapd
+  # run
+  NESTED_BUILD_SNAPD_FROM_CURRENT: true
+
+  # don't use cloud-init, that will be a separate test, we only use sys-user
+  # assertions to create the user for this test
+  NESTED_USE_CLOUD_INIT: false
+
+  # sign all the snaps we build for the image with fakestore
+  NESTED_SIGN_SNAPS_FAKESTORE: true
+
+  # use the testrootorg auto-import assertion
+  # TODO: commit the Go code used to create this assertion from the json file
+  NESTED_CUSTOM_AUTO_IMPORT_ASSERTION: $TESTSLIB/assertions/developer1-auto-import.assert
+
+  # two variants, for signed and secured grades
+  MODEL_GRADE/secured: secured
+  MODEL_GRADE/signed: signed
+
+  NESTED_CUSTOM_MODEL: $TESTSLIB/assertions/developer1-20-${MODEL_GRADE}.model
+  NESTED_IMAGE_ID: testkeys-${MODEL_GRADE}
+
+  # for the fake store
+  NESTED_FAKESTORE_BLOB_DIR: $(pwd)/fake-store-blobdir
+  NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL: http://localhost:11028
+
+  # unset this otherwise ubuntu-image complains about overriding the channel for
+  # a model with grade higher than dangerous when building the image
+  NESTED_CORE_CHANNEL: ""
+
+prepare: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+
+  # setup the fakestore, but don't use it for our snapd here on the host VM, so
+  # tear down the staging_store immediately afterwards so that only the SAS is 
+  # running and our snapd is not pointed at it, ubuntu-image is the only thing 
+  # that actually needs to use the fakestore, and we will manually point it at
+  # the fakestore below using NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL
+  setup_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+  teardown_staging_store
+
+  echo Expose the needed assertions through the fakestore
+  cp "$TESTSLIB"/assertions/developer1.account "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+  cp "$TESTSLIB"/assertions/developer1.account-key "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+
+  # modify and repack gadget snap to add a defaults section and use our own
+  # prepare-device hook to use the fakedevicesvc
+  mkdir extra-snaps
+
+  # Get the snakeoil key and cert for signing gadget assets (shim)
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  snap download --basename=pc --channel="20/edge" pc
+  unsquashfs -d pc-gadget pc.snap
```

Comment:
Is it being deleted? I can't see where the dir pc-gadget is deleted

Reply:
mmm perhaps not, I can delete it too